python/demo/demo_mixed-poisson.py

```diff
@@ -109,88 +104,296 @@
 
 import numpy as np
 
-from basix.ufl import element, mixed_element
-from dolfinx import default_real_type, fem, io, mesh
-from dolfinx.fem.petsc import LinearProblem
-from ufl import Measure, SpatialCoordinate, TestFunctions, TrialFunctions, div, exp, inner
+import dolfinx.fem.petsc
+from basix.ufl import element
+from dolfinx import fem, la, mesh
+from dolfinx.fem.petsc import discrete_gradient, interpolation_matrix
+from dolfinx.mesh import CellType, create_unit_square
+from ufl import (
+    Measure,
+    SpatialCoordinate,
+    TestFunction,
+    TrialFunction,
+    div,
+    exp,
+    inner,
+)
+
+# Solution scalar (e.g., float32, complex128) and geometry (float32/64)
+# types
+dtype = dolfinx.default_scalar_type
+xdtype = dolfinx.default_real_type
+# -
 
-msh = mesh.create_unit_square(MPI.COMM_WORLD, 32, 32, mesh.CellType.quadrilateral)
+# Create mesh a mesh in two dimensions. The iterative solver constructed
+# later requires special construction that is specific to two
+# dimensions. Application in three-dimensions would require a number of
+# changes to the linear solver.
 
+# +
+msh = create_unit_square(MPI.COMM_WORLD, 48, 48, CellType.quadrilateral, dtype=xdtype)
+# -
+#
+# Here we construct compatible function spaces for the mixed Poisson
+# problem. The `V` Raviart-Thomas ($\mathbb{RT}$) space is a vector-valued $H({\rm
+# div})$ conforming space. The `W` space is a space of discontinuous
+# Lagrange function of degree `k`.
```

Comment:
Repetition from above (Line 81-82), maybe we should remove the first mention and introduce the stable pair here?